Source/Core/VideoCommon/LightingShaderGen.h

```diff
@@ -62,7 +62,8 @@ static void GenerateLightShader(T& object, LightingUidData& uid_data, int index,
 		case LIGHTATTN_NONE:
 		case LIGHTATTN_DIR:
 			object.Write("ldir = normalize(" LIGHT_POS".xyz - pos.xyz);\n", LIGHT_POS_PARAMS(index));
-			object.Write("attn = 1.0f;\n");
+			object.Write("attn = 1.0;\n");
+			object.Write("if (length(ldir) == 0.0)\n\t ldir = float3(1.0, 1.0, 1.0);\n");
```

Comment:
0x1p60 is a hexfloat... something like exp2(60).

I'm not really worried about the performance; my only concern would be that GPUs might not handle infinity predictably; if you don't think that's an issue on new cards, it's fine as-is.